sentry_sdk/profiler.py

```diff
@@ -336,19 +338,25 @@ def get_frame_name(frame):
 class Profile(object):
     def __init__(
         self,
-        scheduler,  # type: Scheduler
         transaction,  # type: sentry_sdk.tracing.Transaction
         hub=None,  # type: Optional[sentry_sdk.Hub]
+        scheduler=None,  # type: Optional[Scheduler]
     ):
         # type: (...) -> None
-        self.scheduler = scheduler
-        self.transaction = transaction
```

Comment:
Keeping a reference to the transaction creates a reference cycle as the transaction must keep a reference to the profile. Instead of keeping a reference to the transaction, only pull out the necessary attributes.